pkg/forwarder/transactions_file_storage.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package forwarder
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const retryTransactionsExtension = ".retry"
+const retryFileFormat = "2006_01_02__15_04_05_"
+
+type transactionsFileStorage struct {
+	serializer         *TransactionsSerializer
+	storagePath        string
+	maxSizeInBytes     int64
+	filenames          []string
+	currentSizeInBytes int64
+}
+
+func newTransactionsFileStorage(
+	serializer *TransactionsSerializer,
+	storagePath string,
+	maxSizeInBytes int64) (*transactionsFileStorage, error) {
+
+	if err := os.MkdirAll(storagePath, 0755); err != nil {
+		return nil, err
+	}
+
+	storage := &transactionsFileStorage{
+		serializer:     serializer,
+		storagePath:    storagePath,
+		maxSizeInBytes: maxSizeInBytes,
+	}
+
+	if err := storage.reloadExistingRetryFiles(); err != nil {
+		return nil, err
+	}
+	return storage, nil
+}
+
+// Serialize serializes transactions to the file system.
+func (s *transactionsFileStorage) Serialize(transactions []Transaction) error {
```

Comment:
Unless I miss something `transactionsFileStorage` is only use inside the `forwarder` and doesn't match any public interface, no ? If so, we should make all its method private.

Reply:
Code updated